File: apps/engine/app/routers/health.py
from fastapi import APIRouter, Depends
from sqlalchemy import text
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_session

router = APIRouter(tags=["health"])


@router.get("/health")
async def health(session: AsyncSession = Depends(get_session)):
    try:
        await session.execute(text("SELECT 1"))
        return {"status": "ok", "db": "connected"}
    except Exception:
        from fastapi import Response
        return Response(
            content='{"status":"error","db":"unavailable"}',
            status_code=503,
            media_type="application/json",
        )

File: apps/api/src/routes/health.ts
import type { FastifyInstance } from 'fastify'
import { getPool } from '../lib/db.js'

export async function healthRoutes(app: FastifyInstance) {
  app.get('/health', async (_req, reply) => {
    try {
      await getPool().query('SELECT 1')
      return reply.send({ status: 'ok', db: 'connected' })
    } catch {
      return reply.status(503).send({ status: 'error', db: 'unavailable' })
    }
  })
}
